lambdaclass/mirra_backend
# [GH-399] Deal damage over time

## Motivation

We need to implement the damage over time executions for effects, following the guidelines found in [the design document](https://www.notion.so/lambdaclass/Executions-dd3daa21250848f99b4e6aa6bd2ec468?pvs=4)

Closes https://github.com/lambdaclass/afk_gacha_game/issues/399

## Summary of changes

- Add `executions_over_time` to effects
- Process the executions over time when a skill that has one of them is casted. 
- Add a test for this new behaviour

## How to test it?

Check the tests and play with the skills cooldowns, DoT duration and period to check that the behaviour is the expected one.

## Checklist
